Unable to build frontend when Arches is installed with poetry, pipenv or uv

When loading templates from the Django server, the [path lookup utility function](https://github.com/archesproject/arches/blob/d559dfd2328037e60eb265876ca353a95c20275d/webpack/webpack-utils/build-filepath-lookup.js#L30-L33) ignores any path containing a dot ([#11094](https://github.com/archesproject/arches/issues/11094)).

This has the unintended side effect of preventing frontend build scripts from locating JavaScript files stored within an Arches instance installed in a `.venv`directory, which is favoured by the following Python dependency managers:

 - https://python-poetry.org/docs/configuration/#virtualenvsin-project
 - https://docs.astral.sh/uv/pip/environments/#using-a-virtual-environment
 - https://pipenv.pypa.io/en/latest/virtualenv.html

If Arches is installed in a virtual environment containing a dot, `npm run build_development` fails with the following error:

```bash
root@670418858a9a:/app# uv run npm run build_development

> build_development
> npm run eslint:check && npm run ts:check && cross-env NODE_OPTIONS=--max-old-space-size=2048 webpack --config ./webpack/webpack.config.dev.js


> eslint:check
> eslint **/src


> ts:check
> vue-tsc --noEmit

Data imported from .frontend-configuration-settings.json: {
  _comment: 'This is a generated file. Do not edit directly.',
  APP_ROOT: '/app/finds',
  APP_RELATIVE_PATH: 'finds',
  ARCHES_APPLICATIONS: [ 'finds' ],
  ARCHES_APPLICATIONS_PATHS: { finds: '/app/finds' },
  SITE_PACKAGES_DIRECTORY: '/app/.venv/lib/python3.11/site-packages',
  PUBLIC_SERVER_ADDRESS: 'http://localhost:8000/',
  ROOT_DIR: '/app/.venv/lib/python3.11/site-packages/arches',
  STATIC_URL: '/static/',
  WEBPACK_DEVELOPMENT_SERVER_PORT: 9000
}
asset js/reports/default.js 160 KiB [emitted] (name: reports/default)
asset css/themes/_project.js 2.27 KiB [compared for emit] (name: css/themes/_project)
asset css/themes/_project.css 597 bytes [compared for emit] (name: css/themes/_project)
Entrypoint reports/default 160 KiB = js/reports/default.js
Entrypoint css/themes/_project 2.85 KiB = css/themes/_project.css 597 bytes css/themes/_project.js 2.27 KiB

ERROR in ./finds/media/js/reports/default.js 
Module not found: Error: Can't resolve 'templates/views/report-templates/default.htm' in '/app/finds/media/js/reports'
resolve 'templates/views/report-templates/default.htm' in '/app/finds/media/js/reports'
  Parsed request is a module
  using description file: /app/package.json (relative path: ./finds/media/js/reports)
    Field 'browser' doesn't contain a valid alias configuration
    resolve as module
      looking for modules in /app/node_modules
        /app/node_modules/templates doesn't exist

ERROR in ./finds/media/js/reports/default.js 
Module not found: Error: Can't resolve 'viewmodels/report' in '/app/finds/media/js/reports'
resolve 'viewmodels/report' in '/app/finds/media/js/reports'
  Parsed request is a module
  using description file: /app/package.json (relative path: ./finds/media/js/reports)
    Field 'browser' doesn't contain a valid alias configuration
    resolve as module
      looking for modules in /app/node_modules
        /app/node_modules/viewmodels doesn't exist

webpack 5.97.1 compiled with 2 errors in 1244 ms
```

This can be remedied by amending [the check for a dotfile](https://github.com/archesproject/arches/blob/d559dfd2328037e60eb265876ca353a95c20275d/webpack/webpack-utils/build-filepath-lookup.js#L30-L33), allowing us to skip dot files (such as `.DS_Store`), while allowing files held within a directory containing a dot:

```javascript
…
        // Ignore dotfiles
        if (Path.basename(file).startsWith('.')) {
            return lookup;
        }
…
```

--- a/webpack/webpack-utils/build-filepath-lookup.js
+++ b/webpack/webpack-utils/build-filepath-lookup.js
@@ -28,7 +28,7 @@ function buildFilepathLookup(path, staticUrlPrefix) {
 
     return getFileList(path).reduce((lookup, file) => {
         // Ignore dotfiles
-        if (file.match(new RegExp(Path.sep + '\\.')) || file.match(/^\./)) {
+        if (Path.basename(file).startsWith('.')) {
             return lookup;
         }
         const extension = file.match(/[^.]+$/).toString();

